src/evo/cshelper.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2014-2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_EVO_CSHELPER_H
+#define BITCOIN_EVO_CSHELPER_H
+
+#include <amount.h>
+#include <primitives/transaction.h>
+#include <sync.h>
+#include <threadsafety.h>
+
+#include <optional>
+#include <string>
+#include <vector>
+
+class BlockValidationState;
+class CBlock;
+class CBlockIndex;
+class CCoinsViewCache;
+class CCreditPoolManager;
+class CDeterministicMNManager;
+class CGovernanceManager;
+class CMasternodeSync;
+class CMNHFManager;
+class CRangesSet;
+class CTransaction;
+class CSporkManager;
+class CTxOut;
+class TxValidationState;
+
+struct CMutableTransaction;
+struct MNListUpdates;
+
+namespace Consensus { struct Params; }
+namespace llmq {
+class CQuorumBlockProcessor;
+class CChainLocksHandler;
+} // namespace llmq
+
+extern RecursiveMutex cs_main;
+
+class ChainstateHelper
```

Comment:
why don't move all of that directly to `CChainState` instead?

Reply:
Because I'd prefer `CChainState` to be as close to upstream as possible, not only to make backports and maintenance easier, but to also maintain a soft barrier between Dash and Bitcoin code.

`ChainstateHelper` lets primarily-Bitcoin code access predominantly Dash code, acting as a vessel that abstracts the inner machinations from `CChainState`, letting us change and update Dash-specific code as we see fit without having to make too many changes to Bitcoin-maintained code.